examples/aggregation.json

```diff
@@ -1,152 +1,136 @@
-              "kind": "changed"
-            },
-            {
-              "component": "kubernetes",
-              "description": "Kubernetes version updated.",
-              "kind": "changed"
-            }
-          ],
-          "components": [
-            {
-              "name": "etcd",
-              "version": "3.2.4"
-            },
-            {
-              "name": "kubernetes",
-              "version": "1.8.2"
-            }
-          ],
-          "dependencies": [],
-          "deprecated": false,
-          "time": "2017-08-24T09:11:00Z",
-          "version": "2.1.0",
-          "wip": false
+      "changelog": [
+        {
+          "component": "etcd",
+          "description": "Etcd version updated.",
+          "kind": "changed"
+        },
+        {
+          "component": "kubernetes",
+          "description": "Kubernetes version updated.",
+          "kind": "changed"
         }
       ],
-      "name": "cloud-config-operator"
-    },
+      "components": [
+        {
+          "name": "etcd",
+          "version": "3.2.4"
+        },
+        {
+          "name": "kubernetes",
+          "version": "1.8.2"
+        }
+      ],
+      "dependencies": [],
+      "deprecated": false,
+      "name": "cloud-config-operator",
+      "time": "2017-08-24T09:11:00Z",
+      "version": "2.1.0",
+      "wip": false
+    }
     {
-      "bundles": [
-        {
-          "changelog": [
-            {
-              "component": "calico",
-              "description": "Calico version updated.",
-              "kind": "changed"
-            },
-            {
-              "component": "kubernetes",
-              "description": "Kubernetes version requirements changed due to calico update.",
-              "kind": "changed"
-            }
-          ],
-          "components": [
-            {
-              "name": "calico",
-              "version": "2.0.0"
-            },
-            {
-              "name": "kube-dns",
-              "version": "1.0.0"
-            }
-          ],
-          "dependencies": [
-            {
-              "name": "kubernetes",
-              "version": "<= 1.8.x"
-            }
-          ],
-          "deprecated": false,
-          "time": "2017-08-25T16:40:00Z",
-          "version": "2.0.0",
-          "wip": false
+      "changelog": [
+        {
+          "component": "calico",
+          "description": "Calico version updated.",
+          "kind": "changed"
+        },
+        {
+          "component": "kubernetes",
+          "description": "Kubernetes version requirements changed due to calico update.",
+          "kind": "changed"
+        }
+      ],
+      "components": [
+        {
+          "name": "calico",
+          "version": "2.0.0"
+        },
+        {
+          "name": "kube-dns",
+          "version": "1.0.0"
+        }
+      ],
+      "dependencies": [
+        {
+          "name": "kubernetes",
+          "version": "<= 1.8.x"
```

Comment:
Are we doing the version evaluation or kubernetes? `<= 1.8.x` is really dangerous IMO because `x` gets you in trouble as you don't have definite control over the version any more. `<= 1.8.0` would do the same thing here without introducing a wildcard to your version.

Reply:
You are probably right. This is just an example though to give an impression of how it looks like and how it can work. This is not a guide of best practices. We definitely have to learn what makes sense over time. 

The version defined for the dependency is set by us manually when working on e.g. an operator. In this case you would use a component that requires Kubernetes somehow to be able to reconcile guest clusters properly. You are then responsible for defining which version of Kubernetes your operator's implementation should depend on. Maybe check the tests for the dependency type in the code.

Reply:
- https://godoc.org/github.com/giantswarm/versionbundle#pkg-constants
- https://godoc.org/github.com/giantswarm/versionbundle#Dependency

Reply:
Ack, that makes sense to me. I agree on the learning over time part, we should definitely have a lunch talk once this is put into place to get everyone in the team on kind of the same page about how we want to version things.

Reply:
Yeah you're right. I am not a smart man.